core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/pythonworker/PythonProxyServer.scala

```diff
@@ -106,6 +106,9 @@ private class AmberProducer(
     if (dataHeader.payloadType == EndOfUpstream().getClass.getSimpleName) {
       assert(root.getRowCount == 0)
       outputPort.sendTo(to, MarkerFrame(EndOfUpstream()))
+    } else if (dataHeader.payloadType == State().getClass.getSimpleName) {
+      assert(root.getRowCount == 1)
+      outputPort.sendTo(to, MarkerFrame(State().fromTuple(ArrowUtils.getTexeraTuple(0, root))))
```

Comment:
major: so you are converting arrow data into a tuple and then into a state. This means a state must be convertible from a tuple. I am against this idea. please make state's serialization and deserialization independent from Tuple.

Reply:
Let's keep the assumption that a state must be convertible from a tuple for now. I don't see a use case that they should be different, and we will introduce operators to convert tuple to state and state to tuple. If we don't make this assumption, those operators won't work.